go/extractor/extractor.go

```diff
@@ -773,18 +773,25 @@ func (extraction *Extraction) extractFileInfo(tw *trap.Writer, file string, isDu
 	var parentLbl trap.Label
 
 	for i, component := range components {
+		// displayPath is the same as rawPath except for root directories: if
+		// rawPath is "" then displayPath is "/"; if rawPath is "C:" then
+		// displayPath is "C:/".
+		var rawPath, displayPath string
 		if i == 0 {
-			if component == "" {
-				path = "/"
+			rawPath = component
+			if component == "" || regexp.MustCompile(`^[A-Za-z]:$`).MatchString(component) {
```

Comment:
Am I guessing correctly that at this point we always deal with absolute paths, so the else branch will never run in practice? Otherwise, I guess there is a corner case of a relative `x:` path pointing to a file on Linux leading to a directory `x:/` displayPath...

Reply:
To be honest I don't know. This part of the code hasn't been touched for 4 years. I'll try to look into it.

Reply:
I think we're supposed to emit file elements in the CodeQL DB on absolute paths only, hence my guess